### ISSUE TITLE
Error when applying SoftTreeSupLoss to custom model/dataset

### ISSUE THREAD

**Issue by akshatshreemali** [alvinwan/neural-backed-decision-trees#6]
Hey,
Thanks for sharing your code.
I have a pretrained pytorch(Fastai) model on my custom dataset and when i am trying to run the below code on my dataset. I get an assertion error.

The dataset I have is loaded using a dataloader having 3 classes
from nbdt.loss import SoftTreeSupLoss
criterion = nn.CrossEntropyLoss()
criterion = SoftTreeSupLoss(dataset='CIFAR10', criterion=criterion) -- originial
criterion = SoftTreeSupLoss(dataset=data, criterion=criterion) -- custom

Also, do I have to train the whole model again or can just pass this new loss function to that model and get the inference?

Any help would be really appreciated :)

**Comment by alvinwan** [comment]
Code looks good. What's the error? You can just run the model as an NBDT. However, fine-tuning with the SoftTreeSupLoss will improve accuracy drastically.

**Comment by akshatshreemali** [comment]
Hey
thanks for your reply
everytime i try to run the code for my custom dataset
i get an "assertion error"
below is the code:
from nbdt.loss import SoftTreeSupLoss
criterion = nn.CrossEntropyLoss()
criterion = SoftTreeSupLoss(dataset=data['train'], criterion=criterion)

where data['train'] is a dataloader for my dataset

Sorry for the confusion but please let me know if i am missing anything

**Comment by alvinwan** [comment]
I see, just modify the `nbdt.utils.DATASETS` constant to include your custom dataset. Also, dataset section of the README will be helpful: https://github.com/alvinwan/neural-backed-decision-trees#dataset

**Comment by akshatshreemali** [comment]
Hi,
thanks a lot for your reply and time.
I modified the utils.py and modified the datasets too for my custom dataset.
However, can you guide me to execute it?
The reason being, after modifying both the python files I get the same error.
After modifying both files, is their a different way to run these files?

**Comment by alvinwan** [comment]
Try steps 3 and 4 (I just added these) in [the README "datasets" section](https://github.com/alvinwan/neural-backed-decision-trees/blob/master/README.md#dataset). It now elucidates which constants in `nbdt.utils` to modify.
